code/src/main/kotlin/com/expediagroup/sdk/v2/core/apache/util/ApacheHttpTransportUtil.kt

```diff
@@ -1,35 +1,66 @@
 package com.expediagroup.sdk.v2.core.apache.util
 
-import com.expediagroup.sdk.v2.core.trait.common.IdTrait
 import com.expediagroup.sdk.v2.core.trait.configuration.ClientConfiguration
 import com.google.api.client.http.apache.v2.ApacheHttpTransport
-import java.util.*
-import java.util.concurrent.ConcurrentHashMap
-import java.util.concurrent.ConcurrentMap
 
+/**
+ * Creates an `ApacheHttpTransport` instance based on the provided `ClientConfiguration`.
+ *
+ * @param configuration The `ClientConfiguration` used to configure the `ApacheHttpTransport` instance.
+ * @return A new instance of `ApacheHttpTransport` configured according to the provided `ClientConfiguration`.
+ */
 fun createApacheHttpTransport(configuration: ClientConfiguration): ApacheHttpTransport =
-    ApacheHttpTransport(getSingletonHttpClient(configuration))
+    ApacheHttpTransport(createHttpClient(configuration))
 
+/**
+ * Retrieves a singleton instance of `ApacheHttpTransport` based on the provided `ClientConfiguration`.
+ *
+ * @param configuration The `ClientConfiguration` used to configure the `ApacheHttpTransport` instance.
+ * @return An instance of `ApacheHttpTransport`.
+ */
```

Comment:
I see we have an external dependency for the singleton here `ClientConfiguration`. What happens if we try to initialize another instance with a different  configurations? This code will return the already built instance with the old configurations, right? 
I see that we had something similar to what I'm mentioning here in a previous revision. can we do something like 
```kotlin
private object ApacheHttpTransportCache {
    private val transportMap = ConcurrentHashMap<ClientConfiguration, ApacheHttpTransport>()

    fun getInstance(configuration: ClientConfiguration): ApacheHttpTransport {
        return transportMap.computeIfAbsent(configuration) {
            createApacheHttpTransport(it)
        }
    }
}
```

Reply:
Done.